desktop/renderer-app/src/components/MainPageLayoutContainer/index.tsx

```diff
@@ -112,17 +114,37 @@ export const MainPageLayoutContainer: React.FC<MainPageLayoutContainerProps> = (
         }
     };
 
+    const topBarMenu = [
+        {
+            key: "github",
+            icon: <SVGGithub />,
+            route: "https://github.com/netless-io/flat/",
+        },
+    ];
+
+    const onClickTopBarMenu = (mainPageTopBarMenuItem: MainPageTopBarMenuItem): void => {
+        void shell.openExternal(mainPageTopBarMenuItem.route);
+    };
+
+    const onClickWindowsSystemBtn = (winSystemBtn: WindowsSystemBtnItem): void => {
+        ipcAsyncByMainWindow("set-win-status", { windowStatus: winSystemBtn });
+    };
```

Comment:
Why not put it in `TopBar` instead of writing it on every page?

Reply:
the TopBar just pure component

Reply:
Okey. Perhaps it would be better if we wrapped a component on top of the `pure component` base? Because the logic here is exactly the same.

cc: @crimx